### PR TITLE
Feature/first version

### PR DESCRIPTION
**Pull request by Jean-Serge**

### CODE REVIEW

File: .credo.exs [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+        # Priority values are: `low, normal, high, higher`
+        #
+        {Credo.Check.Design.AliasUsage,
+         [priority: :low, if_nested_deeper_than: 2, if_called_more_often_than: 0]},
+        # You can also customize the exit_status of each check.
+        # If you don't want TODO comments to cause `mix credo` to fail, just
+        # set this value to 0 (zero).
+        #
+        {Credo.Check.Design.TagTODO, false},
+        {Credo.Check.Design.TagFIXME, false},
```
Comment (Eric-P-Fr) [inline .credo.exs:91]: 👍

File: .tool-versions [hunk 2]
```diff
@@ -0,0 +1,2 @@
+erlang 22.3
+elixir 1.10.2
```
Comment (Eric-P-Fr) [inline .tool-versions]: on doit rester dans ces anciennes versions ?

File: lib/sms_factor/webhooks.ex [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  """
+  @type webhook_type() :: String.t()
+
+  @type webhook_params() :: %{webhook: %{type: webhook_type(), url: String.t()}}
+
+  @spec create_webhook(Tesla.Client.t(), webhook_params()) :: Tesla.Env.result()
+  def create_webhook(client, params), do: Tesla.post(client, "/webhook", params)
+
+  @spec retrieve_webhooks(Tesla.Client.t()) :: Tesla.Env.result()
+  def retrieve_webhooks(client), do: Tesla.get(client, "/webhook")
```
Comment (creusefond) [inline lib/sms_factor/webhooks.ex:23]: Why "retrieve" instead of "get"?

File: lib/sms_factor/middleware/parse_response.ex [hunk 4]
```diff
@@ -0,0 +1,39 @@
+defmodule SMSFactor.Middleware.ParseResponse do
+  @moduledoc false
+
+  @behaviour Tesla.Middleware
+
+  @impl Tesla.Middleware
+  def call(env, next, _opts) do
+    # ? Maybe we should also handle Tesla errors
+    with {:ok, response_env} <- Tesla.run(env, next) do
+      handle_successfull_request(response_env)
```
Comment (creusefond) [inline lib/sms_factor/middleware/parse_response.ex:10]: there is only one "l" in "successful"

**Review by creusefond — APPROVED** [review]
Simple, cool, thanks !

File: lib/sms_factor/send_sms.ex [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  @type send_campaign_params() :: %{sms: %{atom() => any()}}
+
+  @typedoc """
+  Params for sending SMS.
+
+  - `token` **(required if not in headers)** : Your token
+  - `text` **(required)** : Your message
+  - `to` **(required)** : Your destination
+  - `pushtype` : The push type (alert or marketing)
+  - `delay` : Sending date Y-m-d H:i:s
```
Comment (Eric-P-Fr) [inline lib/sms_factor/send_sms.ex:52]: pas de timezone, dommage que l'API ne le gère pas. Faudra demander si ils le géreront un jour et dans quel tz on doit fournir ce "delay"

File: lib/sms_factor/webhooks.ex (hunk 3 above)
Comment (Jean-Serge) [reply lib/sms_factor/webhooks.ex:23]: I wanted to be as close as possible to the API, including their wording.

**Review by Zgore14 — APPROVED** [review]

**Review by Eric-P-Fr — APPROVED** [review]

**Pull request merged**